### PR TITLE
[Quantization] Make annotation util functions return annotated nodes

### PR DESCRIPTION
**Pull request by kimishpatel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107107
* #107900
* #107899
* __->__ #107106
* #107105

Summary:
Having annotation functions return nodes that are annotated is useful
specifically for adding "quantization_tag" to those nodes

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D48488415](https://our.internmc.facebook.com/intern/diff/D48488415)

### CODE REVIEW

File: torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py [hunk 1]
```diff
@@ -66,6 +67,8 @@ def _annotate_linear(
     bias_qspec = get_bias_qspec(quantization_config)
     for partitions in module_partitions.values():
         for p in partitions:
+            print(p.nodes)
```
Comment (jerryzh168) [inline torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py:70]: please remove print before landing

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
please update the return type for annotate functions in xnnpack_quantizer.py as well

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48488415).

**Review by jerryzh168 — APPROVED** [review]

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48488415).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48488415).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48488415).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48488415).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48488415).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48488415).

**Pull request closed without merging**